Source/TaskDelegate.swift

```diff
@@ -41,29 +41,23 @@ open class TaskDelegate: NSObject {
 
     var task: URLSessionTask? {
         set {
-            taskLock.lock(); defer { taskLock.unlock() }
-            _task = newValue
-        }
-        get {
-            taskLock.lock(); defer { taskLock.unlock() }
-            return _task
+            protectedTask.unsafeValue = newValue
+            reset()
         }
+        get { return protectedTask.unsafeValue }
     }
     var initialResponseTime: CFAbsoluteTime?
     var credential: URLCredential?
     var metrics: AnyObject? // URLSessionTaskMetrics
 
-    private var _task: URLSessionTask? {
-        didSet { reset() }
-    }
-    private let taskLock = NSLock()
+    private let protectedTask: Protector<URLSessionTask?>
```

Comment:
Is the `protected` prefix just meant to make it more clear throughout the code that it's actually wrapped in a `Protector`? Is it overkill?